ConsoleProject/src/services/TextureConverter.cs

```diff
@@ -30,6 +30,19 @@ public class TextureConverter {
             }
         }
 
+        public List<Texture> ParseNewFile(string[] fileList) {
+            if (fileList != null) { // is this IF required?
+                try {
+                     // TODO only reads first file
+                    return ParseFile(fileList[0]);
+                } catch (InvalidDataException ex) {
+                    Console.WriteLine(ex);
+                    return null;
```

Comment:
Avoid returning with null. Returning with null always causes a null check in the caller layer. Usually only the not-null case is handled, without the else branch. 

I changed the original null return to exception throwing. An exception must be handled, unlike a null return.

Reply:
Here I would remove the try catch, and move it to the caller layer. Let the exception propagate upward.